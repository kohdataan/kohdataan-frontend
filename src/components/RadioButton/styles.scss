@import '../../styles/colors.scss';

/* Customize the label (the radio button) */
.radio-button {
  display: block;
  position: relative;
  padding-left: 35px;
  margin-bottom: 20px;
  margin-top: 20px;
  cursor: pointer;
  font-size: 16px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

/* Hide the browser's default radio button */
.radio-button input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}

/* Create a custom radio button */
.radio-button-checkmark {
  position: absolute;
  top: 0;
  left: 0;
  height: 25px;
  width: 25px;
  background-color: #fff;
  border: solid 2px #3a3a3a;
  border-radius: 50%;
}

/* Create the indicator (the dot/circle - hidden when not checked) */
.radio-button-checkmark:after {
  content: "";
  position: absolute;
  display: none;
}

/* Show the indicator (dot/circle) when checked */
.radio-button input:checked ~ .radio-button-checkmark:after {
  display: block;
}

/* Style the indicator (dot/circle) */
.radio-button .radio-button-checkmark:after {
  top: 3px;
  left: 3px;
  width: 15px;
  height: 15px;
  border-radius: 50%;
  background: $orange;
}