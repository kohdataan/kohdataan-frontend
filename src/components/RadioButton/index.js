import React, { memo } from 'react'
import PropTypes from 'prop-types'
import './styles.scss'

const RadioButton = props => {
  const { name, labelClassName, inputClassName, value, onChange, label, checked } = props

  return (
    <label htmlFor={label} className={`radio-button ${labelClassName}`}>
      {label}
      <input
        type="radio"
        name={name}
        id={label}
        value={value}
        onChange={onChange}
        className={inputClassName}
        checked={checked}
        onKeyPress={onChange}
        tabIndex={0}
      />
      <span className="radio-button-checkmark" />
    </label>
  )
}

RadioButton.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.string,
  inputClassName: PropTypes.string,
  labelClassName: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  checked: PropTypes.bool,
}

RadioButton.defaultProps = {
  value: '',
  inputClassName: '',
  labelClassName: '',
  checked: false,
}

export default memo(RadioButton)
