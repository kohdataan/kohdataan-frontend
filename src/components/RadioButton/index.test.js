import React from 'react'
import { create } from 'react-test-renderer'
import RadioButton from './index'

describe('Button container', () => {
  test('it matches the snapshot', () => {
    const component = create(
      <RadioButton label="test" name="test" onChange={() => {}} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
