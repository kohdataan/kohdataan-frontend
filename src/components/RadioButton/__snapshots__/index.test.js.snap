// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button container it matches the snapshot 1`] = `
<label
  className="radio-button "
  htmlFor="test"
>
  test
  <input
    checked={false}
    className=""
    id="test"
    name="test"
    onChange={[Function]}
    onKeyPress={[Function]}
    tabIndex={0}
    type="radio"
    value=""
  />
  <span
    className="radio-button-checkmark"
  />
</label>
`;
